sdk/resource_locator.go

```diff
@@ -48,28 +48,11 @@ func NewResourceLocator(url string) (*ResourceLocator, error) {
 
 func NewResourceLocatorFromReader(reader io.Reader) (*ResourceLocator, error) {
 	rl := &ResourceLocator{}
-	oneByte := make([]byte, 1)
-
-	_, err := reader.Read(oneByte)
-	if err != nil {
-		return rl, err
-	}
-	rl.protocol = urlProtocol(oneByte[0])
-
-	_, err = reader.Read(oneByte)
-	if err != nil {
-		return rl, err
-	}
-
-	l := oneByte[0]
-	body := make([]byte, l)
-	_, err = reader.Read(body)
+	err := rl.readResourceLocator(reader)
 	if err != nil {
-		return rl, err
+		return nil, err
```

Comment:
No concerns at all, just highlighting this is a behavior change (I was trying to match the existing behavior as close as I can).